scico/functional/_norm.py

```diff
@@ -477,3 +479,110 @@ def prox(
         svdU, svdS, svdV = snp.linalg.svd(v, full_matrices=False)
         svdS = snp.maximum(0, svdS - lam)
         return svdU @ snp.diag(svdS) @ svdV
+
+
+class TV2DNorm(Functional):
+    r"""The anisotropic total variation (TV) norm.
+
+    For a :math:`M \times N` matrix, :math:`\mb{A}`, by default,
+
+    .. math::
+           \norm{\mb{A}}_{\text{TV}} = \sum_{n=1}^N \sum_{m=1}^M
+           \abs{\nabla{A}_{m,n}} \;.
+
+    The proximal operator of this norm is currently only defined for 2
+    dimensional data.
+
+    For `BlockArray` inputs, the TV norm follows the reduction rules
+    described in :class:`BlockArray`.
+    """
+
+    has_eval = True
+    has_prox = True
+
+    def __init__(self, tau: float = 1.0):
+        r"""
+        Args:
+            tau: Parameter :math:`\tau` in the norm definition.
+        """
+        self.tau = tau
```

Comment:
Does this play a useful role? The base `Functional` already handles multiplication by a scalar.

Reply:
Sorry I didn't understand. Do you mean the role of the `tau` parameter? It affects the threshold in the `prox` function.

Reply:
But it just multiplies the prox scaling parameter, no? So what does it add over that parameter?

Reply:
@bwohlberg , sure it does scales the prox parameter in the `prox()` function, but it is also used in the `__call__` function to evaluate the Norm to scale the output, right?